client-v2/src/reducers/clipboardReducer.ts

```diff
@@ -21,15 +24,51 @@ const clipboard = (
       return payload;
     }
     case REMOVE_CLIPBOARD_ARTICLE_FRAGMENT: {
-      return state.filter(id => id !== action.payload.articleFragmentId);
+      if (action.payload.clipboardType === 'fronts') {
+        return {
+          ...state,
+          ...{
+            frontsClipboard: state.frontsClipboard.filter(
+              id => id !== action.payload.articleFragmentId
+            )
+          }
+        };
+      }
+      if (action.payload.clipboardType === 'editions') {
+        return {
+          ...state,
+          ...{
+            editionsClipboard: state.editionsClipboard.filter(
+              id => id !== action.payload.articleFragmentId
+            )
+          }
+        };
+      }
+      return state;
     }
+
     case INSERT_CLIPBOARD_ARTICLE_FRAGMENT: {
-      return insertAndDedupeSiblings(
-        state,
-        [action.payload.articleFragmentId],
-        action.payload.index,
-        articleFragmentsSelector(prevSharedState)
-      );
+      if (action.payload.clipboardType === 'fronts') {
```

Comment:
I know there is a lot of code duplication here but I sort of like it because it makes the intention and what is being updated very clear.